src/nano-network-api.js

```diff
@@ -339,41 +110,82 @@ export class NanoNetworkApi {
         return balances;
     }
 
+    // To support un-updated client code
+    async connectPico(addresses) {
+        return this.getBalance(addresses);
+    }
+
     /**
      * @param {string} address
      * @returns {string | boolean}
      */
     async getAccountTypeString(address) {
         const account = (await this._getAccounts([address]))[0];
-
-        if (!account) return 'basic';
-
-        // See Nimiq.Account.Type
-        switch (account.type) {
-            case Nimiq.Account.Type.BASIC: return 'basic';
-            case Nimiq.Account.Type.VESTING: return 'vesting';
-            case Nimiq.Account.Type.HTLC: return 'htlc';
-            default: return false;
-        }
+        return Nimiq.Account.Type.toString(account.type);
     }
 
+    /**
+     * @param {string[]} addresses
+     * @param {Map} [knownReceipts] A map with the tx hash as key and the blockhash as value (both base64)
+     * @param {uint} [fromHeight]
+     */
     async requestTransactionHistory(addresses, knownReceipts, fromHeight) {
         if (!(addresses instanceof Array)) addresses = [addresses];
+        await this._consensusEstablished;
 
-        let results = await Promise.all(addresses.map(address => this._requestTransactionHistory(address, knownReceipts.get(address), fromHeight)));
-
-        // txs is an array of objects of arrays, which have the format {transaction: Nimiq.Transaction, header: Nimiq.BlockHeader}
-        // We need to reduce this to usable simple tx objects
-
-        // Construct arrays with their relavant information
-        let txs = results.map(r => r.transactions);
-        let removedTxs = results.map(r => r.removedTxHashes);
-        let unresolvedTxs = results.map(r => r.unresolvedReceipts);
+        const receipts =
+            // 1. Get all receipts for all addresses
+            (await Promise.all(addresses.map(address => this._client.getTransactionReceiptsByAddress(address))))
+            // 2. Flatten array
+            .reduce((flat, it) => flat.concat(it), [])
+            // 3. Remove duplicates
+            .filter((receipt, index, receipts) => receipts.indexOf(receipt) === index);
+
+        const newReceipts = receipts
+            // 4. Remove knownReceipts
+            .filter(receipt => {
+                if (receipt.blockHeight < fromHeight) return false;
+                const txHash = receipt.transactionHash.toBase64();
+                if (knownReceipts.has(txHash)) {
+                    // Check if block has changed
+                    return receipt.blockHash.toBase64() !== knownReceipts.get(txHash);
+                }
+                return true;
+            })
+            // 5. Sort in reverse, to resolve recent transactions first
```

Comment:
If the browser node is running into rate limiting by peers, I would rather it can't get older transactions than the most recent transactions. So to fetch the recent ones first (before potentially being rate limited), the array is reversed.